controller/src/controller.rs

```diff
@@ -11,14 +11,20 @@ use kube::runtime::reflector::Store;
 use kube::ResourceExt;
 use opentelemetry::global;
 use snafu::ResultExt;
-use std::collections::{BTreeMap, HashMap};
 use std::env;
+use std::{
+    collections::{BTreeMap, HashMap},
+    convert::TryInto,
+};
 use tokio::time::{sleep, Duration};
 use tracing::{event, instrument, Level};
 
 // Defines the length time after which the controller will take actions.
 const ACTION_INTERVAL: Duration = Duration::from_secs(2);
 
+// Defines environmental variable name used to fetch max concurrent update number.
```

Comment:
Nit: You missed in replacement:

```suggestion
// Defines environment variable name used to fetch max concurrent update number.
```